Wikipedia/Code/StorageAndSyncingSettingsViewController.swift

```diff
@@ -0,0 +1,213 @@
+struct SettingsSection {
+    let headerTitle: String?
+    let footerText: String?
+    let items: [SettingsItem]
+}
+
+struct SettingsItem {
+    let disclosureType: WMFSettingsMenuItemDisclosureType?
+    let itemType: StorageAndSyncingSettingsItemType
+    let title: String?
+    let iconName: String?
+    let iconColor: UIColor?
+    let iconBackgroundColor: UIColor?
+    let isSwitchOn: Bool?
+    let buttonTitle: String?
+}
+
+enum StorageAndSyncingSettingsItemType: Int {
+    case syncSavedArticlesAndLists, showSavedReadingList, eraseSavedArticles, syncWithTheServer
+    
+    public func settingsItem(isSwitchOnByDefault: Bool? = nil) -> SettingsItem {
```

Comment:
This func is essentially an Item constructor, so should be part of the Item struct as something like: `init(for itemType: StorageAndSyncingSettingsItemType, disclosureType: WMFSettingsMenuItemDisclosureType, ...`